extensions/linear/src/components/IssueActions/MilestoneSubmenu.tsx

```diff
@@ -0,0 +1,71 @@
+import { Action, ActionPanel, Color } from "@raycast/api";
+import { useState } from "react";
+
+import { IssueResult } from "../../api/getIssues";
+import { UpdateIssueParams } from "./IssueActions";
+import { MilestoneResult } from "../../api/getMilestones";
+import useMilestones from "../../hooks/useMilestones";
+import { getMilestoneIcon } from "../../helpers/milestones";
+
+export default function MilestoneSubmenu({
+  issue,
+  updateIssue,
+}: {
+  issue: IssueResult;
+  updateIssue: (params: UpdateIssueParams) => void;
+}) {
+  const [load, setLoad] = useState(false);
+  const { milestones, isLoadingMilestones } = useMilestones(issue.project?.id, { execute: load });
+
+  async function setMilestone(milestone: MilestoneResult | null) {
+    const currentMilestone = issue.projectMilestone;
+    updateIssue({
+      animatedTitle: "Setting milestone",
+      payload: { projectMilestoneId: milestone ? milestone.id : null },
+      optimisticUpdate(issue) {
+        return {
+          ...issue,
+          milestone: milestone || undefined,
+        };
+      },
+      rollbackUpdate(issue) {
+        return {
+          ...issue,
+          milestone: currentMilestone,
+        };
+      },
+      successTitle: milestone ? "Set milestone" : `Removed milestone from ${issue.identifier}`,
+      successMessage: milestone ? `"${milestone.name}" added to ${issue.identifier}` : "",
+      errorTitle: "Failed to set milestone",
+    });
+  }
+
+  return (
+    <ActionPanel.Submenu
+      title="Set Milestone"
+      icon={{ source: "linear-icons/milestone.svg", tintColor: Color.PrimaryText }}
```

Comment:
`Color.PrimaryText` is already the default color.

```suggestion
      icon="linear-icons/milestone.svg"
```